src/parse.rs

```diff
@@ -563,10 +563,23 @@ where
                 }))
             }
             Some('{') => {
+                // if parsed input is actually invalid, keep the previous one for rollback
```

Comment:
The approach looks good to me but I think this can be simplified by factoring out the quantifier parsing. Today that is inline inside `try_consume_quantifier_prefix`; but it can be factored into a separate function, allowing the call site to branch on Unicode or not. So this code could look like so:

    if let Some(quantifier) = self.try_consume_braced_quantifier() {
          // Success
    } else if self.flags.unicode {
         return error("Invalid quantifier");
    } else {
        return Ok(None);
    }